beaver/transports/rabbitmq_transport.py

```diff
@@ -28,109 +28,104 @@ def __init__(self, beaver_config, logger=None):
         self._channel = None
         self._count = 0
         self._lines = Queue()
+        self._connection_ok = False
+        self._thread = None
+        self._is_valid = True
         self._connect()
 
     def _on_connection_open(self,connection):
-        self._logger.debug("connection created")
+        self._logger.debug("RabbitMQ: Connection Created")
         self._channel = connection.channel(self._on_channel_open)
 
     def _on_channel_open(self,unused):
-        self._logger.debug("Channel Created")
+        self._logger.debug("RabbitMQ: Channel Created")
         self._channel.exchange_declare(self._on_exchange_declareok,
                                        exchange=self._rabbitmq_config['exchange'],
                                        exchange_type=self._rabbitmq_config['exchange_type'],
                                        durable=self._rabbitmq_config['exchange_durable'])
 
     def _on_exchange_declareok(self,unused):
-        self._logger.debug("Exchange Declared")
+        self._logger.debug("RabbitMQ: Exchange Declared")
         self._channel.queue_declare(self._on_queue_declareok,
                                     queue=self._rabbitmq_config['queue'],
                                     durable=self._rabbitmq_config['queue_durable'],
                                     arguments={'x-ha-policy': 'all'} if self._rabbitmq_config['ha_queue'] else {})
 
     def _on_queue_declareok(self,unused):
-        self._logger.debug("Queue Declared")
+        self._logger.debug("RabbitMQ: Queue Declared")
         self._channel.queue_bind(self._on_bindok,
                                  exchange=self._rabbitmq_config['exchange'],
                                  queue=self._rabbitmq_config['queue'],
                                  routing_key=self._rabbitmq_config['key'])
 
     def _on_bindok(self,unused):
-        self._logger.debug("Exchange to Queue Bind OK")
-        self._is_valid = True;
-        self._logger.debug("Scheduling next message for %0.1f seconds",1)
-        self._connection.add_timeout(1,self._publish_message)
-
+        self._logger.info("RabbitMQ: Connection OK.")
+        self._connection_ok = True
+        self._logger.debug("RabbitMQ: Scheduling regular message transport.")
+        self._connection.add_timeout(1, self._publish_message)
 
     def _publish_message(self):
-        while True:
-            self._count += 0
-            if self._lines.not_empty:
```

Comment:
Queue.not_empty is always Truthy, because it is an Object. This if statement therefore did not do anything...